### PR TITLE
Norm: 25 lines per f(): parse_and_or_list.c.

### PR DESCRIPTION
**Pull request by M5oul**
- #26
This is a beginning to reduce the size of the function.
But, it's still > 25 lines.
Not easy to split.

### CODE REVIEW

File: srcs/parse_input/parse_and_or_list.c [hunk 1]
```diff
@@ -77,3 +68,17 @@ t_and_or_list		*parse_and_or_list(t_token const *tokens)
...
+	t_and_or_list	*result;
+	t_token			**splited;
+	size_t			splited_len;
+
+	set_error(NO_ERROR);
+	if (tokens == NULL)
+		return (NULL);
+	splited = do_the_split(tokens, &splited_len);
+	result = NULL;
+	return (parse_and_or_list_run(result, 0, splited, splited_len));
```
Comment (pilespin) [inline srcs/parse_input/parse_and_or_list.c]: Why don't remove result and put directly NULL?

**Comment by M5oul** [comment]
Pfff, I said it's a beginning. It didn't break anything.
You could continue. I didn't found a solution for 25 lines.

**Comment by n0izn0iz** [comment]
Juste pour clarifier l'issue que j'ai ouverte a rien a voir avec cette PR apart que j'ai vu le pbm grace a la review

**Comment by M5oul** [comment]
Can someone try to reduce this function to 25 lines.
I really don't know how to do so.

**Comment by M5oul** [comment]
I approve last commit.

**Pull request merged**